src/components/common/ens_address/ens_address.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Paper from "@material-ui/core/Paper";
+import InputBase from "@material-ui/core/InputBase";
+import IconButton from "@material-ui/core/IconButton";
+import SearchIcon from "@material-ui/icons/Search";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import { withStyles } from "@material-ui/core/styles";
+import classnames from "classnames";
+import { ethers } from "ethers";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import actions from "actions";
+import { styles } from "./styles";
+import { isEthAddress } from "helpers/validators";
+
+class ENSAddress extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			resolved: "",
+			error: "",
+			isSearching: false,
+			address: ""
+		};
+
+		this.reverseAddress = this.reverseAddress.bind(this);
+	}
+
+  componentDidUpdate = async (prevProps, prevState) => {
+  	const { address } = this.state;
+  	if (address !== prevState.address) {
+  		const provider = ethers.getDefaultProvider();
+  		const { updateENSResolution } = this.props.actions;
+  		updateENSResolution({
+  			ensName: "",
+  			address
+  		});
+  		if (isEthAddress(address)) {
+  			this.setState({
+  				isSearching: true
+  			});
+
+  			console.log(address.indexOf(""));
+
+  			await provider.lookupAddress(address).then(async ensName => {
+  				if (ensName !== null) {
+  					const resolvedAddress = await provider.resolveName(ensName);
+  					if (resolvedAddress === address) {
+  						this.setState({
+  							isSearching: false
+  						});
+  						updateENSResolution({ ensName, address });
+  					}
+  				} else {
+  					this.setState({
+  						error: "Address has no ens name associated to it.",
```

Comment:
use [adex-translations](https://github.com/AdExNetwork/adex-translations)
add the string to `en-US.json` with pull request.

use `import Translate from 'components/translate/Translate'` `(Translate(withStyles(styles)(Connect))`)